File: src/055. Lychrel numbers/055.rb
class Integer
  def is_lychrel?
    num = self
    i = 0
    while i < 50
      num = num + num.to_s.reverse.to_i
      if num == num.to_s.reverse.to_i
        return false
      end
      i += 1
    end
    true
  end
end

counter = 0
1.upto(10000) do |num|
  if num.is_lychrel?
    counter += 1
  end
end
puts counter

File: src/055. Lychrel numbers/055.py
def is_lychrel(num):
    i = 0
    while i < 50:
        num = num + int(str(num)[::-1])
        if num == int(str(num)[::-1]):
            return False
        i += 1
    return True

counter = 0
for i in range(1, 10001):
    if is_lychrel(i):
        counter += 1

print(counter)
